sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuShuffleExchangeExec.scala

```diff
@@ -49,6 +49,10 @@ class GpuShuffleMeta(
   override val childParts: scala.Seq[PartMeta[_]] =
     Seq(GpuOverrides.wrapPart(shuffle.outputPartitioning, conf, Some(this)))
 
+  // propagate possible type conversions on the output attributes of map-side plans to
+  // reduce-side counterparts
+  override protected val useOutputAttributesOfChild: Boolean = true
```

Comment:
I am also confused why this PR changes the behavior of GpuShuffleMeta. Can you elaborate? An example in the PR description would be nice.

Reply:
In most situations, at least one `ShuffleExchangeExec`  will be inserted for each Aggregate logical plan. ShuffleExchange for aggregate takes the internal aggregation buffers along with grouping keys. As we override the data type of aggregation buffer created by `TypedImperativeAggregate` functions,  we also need to apply the same overriding on ShuffleExhange for those buffers.

For instance, we have query ```select a, collect_list(b) from table group by a;```, which physical plan on CPU looks like:
```
ObjectHashAggregate(keys=[a#10], functions=[collect_list(b#11, 0, 0)], output=[a#10, collect_list(b)#17])
+- Exchange hashpartitioning(a#10, 200), true, [id=#13]
   +- ObjectHashAggregate(keys=[a#10], functions=[partial_collect_list(b#11, 0, 0)], output=[a#10, buf#21])
      +- LocalTableScan [a#10, b#11]
```
Among the outputs of the partial aggregate, `buf#21`, the attribute referring to aggregation buffer of CollectList, whose data type in CPU runtime(BinaryType) is different from GPU runtime(Array of ElementType). We need to override it in AggregateMeta. And we also need to propagate the overriding through ShuffleExchange (to the final stage).